modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -428,6 +428,17 @@ protected ImmutableMap.Builder<String, Lambda> addMustacheLambdas() {
     @Override
     public void postProcessModelProperty(CodegenModel model, CodegenProperty property) {
         super.postProcessModelProperty(model, property);
+        if (property.isInnerEnum && property.items != null) {
+            // format maps of inner enums to include the classname eg: Dictionary<string, MapTest.InnerEnum>
+            property.datatypeWithEnum = property.datatypeWithEnum.replace(property.items.datatypeWithEnum, model.classname + "." + property.items.datatypeWithEnum);
+            property.dataType = property.datatypeWithEnum;
+        }
+        if (property.isMap || property.isContainer) {
+            // maps of enums will be marked both isMap and isEnum, correct that now
+            property.isEnum = false;
```

Comment:
How do we handle `Enum` properties ? Now the templates will have false values for `{{#isEnum}}` even if they are Enum.

Reply:
That is the correct behavior. Maps and containers are not enums.

Reply:
🤔  Well, my client is broken . I had  updated template as `List<datatype>`  for  `List<...Enum>`  which was generated as `List<string>`  but not anymore because of this PR change.
 The reason i had updated template was because I had deserialization error in the past if there is no matching enum type for the parameter received, therefore, I had converted `List<Enum>` as `List<datatype>`
  
 I see `enumUknownDefaultCase` [config option](https://github.com/OpenAPITools/openapi-generator/blob/2df68d9359a3d8f7d2b2a98ccf4c18546ce0fc9b/docs/generators/markdown.md)  can be used. 
 But, my question is if this has been implemented in `csharp-netcore` ? If yes, can you point it out?

Reply:
The issue seems to be due to your template. Try using the CodegenProperty#items property to resolve the problem. If you can't figure it out open an issue with steps to reproduce.